dev/benchmarks/test_apps/stocks/lib/stock_state.dart

```diff
@@ -0,0 +1,187 @@
+  final BackupMode backupMode;
+  final bool debugShowGrid;
+  final bool debugShowSizes;
+  final bool debugShowBaselines;
+  final bool debugShowLayers;
+  final bool debugShowPointers;
+  final bool debugShowRainbow;
+  final bool showPerformanceOverlay;
+  final bool showSemanticsDebugger;
+
+  StockConfiguration copyWith({
+    StockMode stockMode,
+    BackupMode backupMode,
+    bool debugShowGrid,
+    bool debugShowSizes,
+    bool debugShowBaselines,
+    bool debugShowLayers,
+    bool debugShowPointers,
+    bool debugShowRainbow,
+    bool showPerformanceOverlay,
+    bool showSemanticsDebugger,
+  }) {
+    return StockConfiguration(
+      stockMode: stockMode ?? this.stockMode,
+      backupMode: backupMode ?? this.backupMode,
+      debugShowGrid: debugShowGrid ?? this.debugShowGrid,
+      debugShowSizes: debugShowSizes ?? this.debugShowSizes,
+      debugShowBaselines: debugShowBaselines ?? this.debugShowBaselines,
+      debugShowLayers: debugShowLayers ?? this.debugShowLayers,
+      debugShowPointers: debugShowPointers ?? this.debugShowPointers,
+      debugShowRainbow: debugShowRainbow ?? this.debugShowRainbow,
+      showPerformanceOverlay: showPerformanceOverlay ?? this.showPerformanceOverlay,
+      showSemanticsDebugger: showSemanticsDebugger ?? this.showSemanticsDebugger,
+    );
+  }
+}
+
+class RouterState extends ChangeNotifier {
+  RouterState();
+
+  bool get debugInitialized {
+    bool answer;
+    assert((){
+      answer = routePath != null;
+      return true;
+    }());
+    return answer;
+  }
+
+  StockRoutePath get routePath => _routePath;
+  StockRoutePath _routePath;
+  set routePath(StockRoutePath value) {
+    if (value != _routePath) {
+      _routePath = value;
+      notifyListeners();
+    }
+  }
+}
+
+class RouterStateScope extends InheritedNotifier<RouterState> {
+  const RouterStateScope({
+    Key key,
+    RouterState routerState,
+    Widget child,
+  }) : super(key: key, notifier: routerState, child: child);
+
+  static RouterState of(BuildContext context) {
+    return context.dependOnInheritedWidgetOfExactType<RouterStateScope>().notifier;
+  }
+}
+
+class _StockConfigurationScope extends InheritedWidget {
+  const _StockConfigurationScope({
+    Key key,
+    this.configuration,
+    Widget child,
+  }) : super(key: key, child: child);
+
+  final StockConfiguration configuration;
+
+  @override
+  bool updateShouldNotify(covariant _StockConfigurationScope oldWidget) {
+    return configuration != oldWidget.configuration;
+  }
+}
+
+class StockStateScope extends StatefulWidget {
+  const StockStateScope({
+    Key key,
+    this.stocks,
+    this.child,
+  }) : super(key: key);
+
+  final StockData stocks;
+  final Widget child;
+
+  static StockState of(BuildContext context) {
+    return context.findAncestorStateOfType<StockState>();
+  }
+
+  static StockData stockDataOf(BuildContext context) {
+    return context.findAncestorWidgetOfExactType<StockStateScope>().stocks;
+  }
```

Comment:
This only works in this toy example because the stock data never changes, right? If it would, you'd want a dependency to rebuild whenever the data changes, no?